Spring-boot-dev-tools is bundled within docker image (causing ClassCastExceptions)

I start seeing #155 with the recent [entur/lamassu@2023-09-20T14-28](https://hub.docker.com/layers/entur/lamassu/2023-09-20T14-28/images/sha256-af9fd9e4e4146244088a01a4535d0274350d6f02c8b4fadca33182ae5cbbc360?context=explore), though no `ENV=DEV` is activated, i.e. in production mode:

```
{"serviceContext":{"service":"lamassu"},"message":"Caught exception in ForkJoinPool\njava.lang.ClassCastException: null\n\tat java.base/jdk.internal.reflect.GeneratedConstructorAccessor189.newInstance(Unknown Source)\n\tat java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)\n\tat java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)\n\tat java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)\n\tat java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:564)\n\tat java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:591)\n\tat java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:689)\n\tat java.base/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)\n\tat java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173)\n\tat java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)\n\tat java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)\n\tat java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:765)\n\tat org.entur.gbfs.GbfsSubscriptionManager.lambda$update$0(GbfsSubscriptionManager.java:68)\n\tat java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)\n\tat java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)\n\tat java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)\n\tat java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)\n\tat java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)\n\tat java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)\nCaused by: java.lang.ClassCastException: null\n","reportLocation":{"filePath":"org.entur.lamassu.leader.FeedUpdater","lineNumber":"95","functionName":"lambda$start$0"},"severity":"WARNING"}
```

As the result is temporary loss of data, this is more serious than just seeing this issue in development.

Inspecting `/app/lib`dir in the container, I see `spring-boot-devtools-2.7.15.jar` in the packaged libs. [pom.xml declares this lib as optional runtime](https://github.com/entur/lamassu/blob/master/pom.xml#L80-L84), but nevertheless, jib packages it within the docker image, which is expected and discussed also in https://github.com/GoogleContainerTools/jib/issues/2336.  

An appropriate way to solve this (by excluding spring-boot-devtools), might be to add https://github.com/GoogleContainerTools/jib-extensions/tree/master/first-party/jib-spring-boot-extension-maven to pom.xml.

--- a/pom.xml
+++ b/pom.xml
@@ -257,6 +257,14 @@
                 <groupId>com.google.cloud.tools</groupId>
                 <artifactId>jib-maven-plugin</artifactId>
                 <version>3.3.2</version>
+                <dependencies>
+                    <!-- Include jib-spring-boot-extension-maven to prevent spring-boot-devtools from being packaged.-->
+                    <dependency>
+                        <groupId>com.google.cloud.tools</groupId>
+                        <artifactId>jib-spring-boot-extension-maven</artifactId>
+                        <version>0.1.0</version>
+                    </dependency>
+                </dependencies>
                 <configuration>
                     <from>
                         <image>eclipse-temurin:17.0.6_10-jdk-alpine</image>
@@ -274,6 +282,11 @@
                             <port>8080</port>
                         </ports>
                     </container>
+                    <pluginExtensions>
+                        <pluginExtension>
+                            <implementation>com.google.cloud.tools.jib.maven.extension.springboot.JibSpringBootExtension</implementation>
+                        </pluginExtension>
+                    </pluginExtensions>
                 </configuration>
             </plugin>
         </plugins>

